File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    count = 0
    visited = [False] * n # [False, False, False]
    # visited = [False for _ in range(0, n)]
    
    def dfs(vertex):
        # 더 이상 도달할 곳이 없는 경우 -> 단순히 반복이 끝난 경우
        visited[vertex] = True
        
        for idx in range(n):
            if computers[vertex][idx] == 1 and visited[idx] == False:
                dfs(idx)
        
    for i in range(n):  # 0부터 n-1까지
        if visited[i] == False:
            dfs(i)
            count += 1
    
    return count

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
import java.io.*;

class Solution {
    
    static boolean[] isVisited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        isVisited = new boolean[n];
        for (int i=0; i<n; i++) {
            if (!isVisited[i]) {
                dfs(i, computers);
                answer++;
            }
        }
        
        return answer;
    }
    
    static void dfs(int idx, int[][] computers) {
        isVisited[idx] = true;    
        
        for (int i=0; i<computers.length; i++) {
            if (computers[idx][i] == 1 && !isVisited[i]) {    // 연결되어있고, 아직 방문하지 않았다면
                dfs(i, computers);
            }
        }
    }
}
